tests/test_validation/config_validation_test.py

```diff
@@ -144,7 +144,16 @@ def test_job_size_failure():
     })
 
     success = validator.validate(configurations).check_success()
-    assert success is False
+
+    return success
+
+
+def test_job_size_failure():
```

Comment:
I think each assert should gets it's own test e.g. test_job_size_failure_for_negative_job_size and test_job_size_failure_for_excessive_job_size
That way if one breaks it's clear in what way it's broken.  Also try to run tests on more of the possible scenarios e.g. when we use a valid job size, when we use an invalid max_objects_per_file but record_count is 0, when we pass in a string as a max_objects_per_file